common/src/main/java/io/novafoundation/nova/common/utils/FlowExt.kt

```diff
@@ -308,15 +315,42 @@ inline fun <T> EditText.bindTo(
             if (text.toString() != inputString) {
                 removeTextChangedListener(textWatcher)
                 setText(inputString)
-                if (hasFocus()) {
-                    setSelection(inputString.length)
+                if (moveSelectionToEndOnInsertion) {
+                    moveSelectionToTheEnd()
                 }
                 addTextChangedListener(textWatcher)
             }
         }
     }
 }
 
+fun ValidatableInputField.observeErrors(
+    flow: Flow<FieldValidationResult>,
+    scope: CoroutineScope,
+) {
+    scope.launch {
+        flow.collect { validationResult ->
+            when (validationResult) {
+                is FieldValidationResult.Ok -> {
+                    hideError()
+                }
+
+                is FieldValidationResult.Error -> {
+                    showError(validationResult.reason)
+                }
+
+                else -> {}
+            }
+        }
+    }
+}
```

Comment:
I think more appropriate place for this is somewhere neraby FieldValidator itself